apps/wallet/src/background/keyring/VaultStorage.test.ts

```diff
@@ -0,0 +1,318 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { describe, expect, it, vi } from 'vitest';
+import Browser from 'webextension-polyfill';
+
+import { VaultStorage } from './VaultStorage';
+import { SESSION_STORAGE } from './storage-utils';
+import * as Bip39 from '_shared/utils/bip39';
+import {
+    testEd25519Serialized,
+    testVault,
+    testVault1,
+} from '_src/test-utils/vault';
+
+describe('VaultStorage', () => {
+    describe('create', () => {
+        it('throws if already initialized', async () => {
+            const vaultStorage = new VaultStorage();
+            vi.spyOn(Browser.storage.local, 'get').mockResolvedValue({
+                vault: {},
+            });
+            await expect(vaultStorage.create('12345')).rejects.toThrow(
+                'Mnemonic already exists, creating a new one will override it. Clear the existing one first.'
```

Comment:
I'd be fine if we just expected to throw but didn't assert on the error message itself.